src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -1136,6 +1139,98 @@ export const GlobalProvider = memo(
+
+            return {
+                x: minX,
+                y: minY,
+                width: maxX - minX,
+                height: maxY - minY,
+            };
+        }, [getNodes]);
+
+        const downloadImage = (dataUrl: string, fileName: string) => {
+            const a = document.createElement('a');
+            a.href = dataUrl;
+            a.download = fileName;
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
+        };
+
+        const [viewportExportPadding] = useViewportExportPadding;
+        const exportViewportScreenshot = useCallback(() => {
+            if (!reactFlowWrapper.current) return;
+
+            const oldViewport = currentReactFlowInstance.getViewport();
+
+            const reactFlowViewport = reactFlowWrapper.current.getBoundingClientRect();
+            const { x, y, width, height } = getNodesBoundingBox();
+            const paddedBoundingBox = {
+                x: x - viewportExportPadding,
+                y: y - viewportExportPadding,
+                width: width + viewportExportPadding * 2,
+                height: height + viewportExportPadding * 2,
+            };
+
+            const exportZoom = Math.min(
+                reactFlowViewport.width / paddedBoundingBox.width,
+                reactFlowViewport.height / paddedBoundingBox.height
+            );
+
+            currentReactFlowInstance.setViewport({
+                x: paddedBoundingBox.x * -1 * exportZoom,
+                y: paddedBoundingBox.y * -1 * exportZoom,
+                zoom: exportZoom,
+            });
+
+            const currentFlowWrapper = reactFlowWrapper.current;
+            if (!(currentFlowWrapper instanceof HTMLElement)) return;
+
+            // wait for the viewport to be updated
+            setTimeout(() => {
+                toPng(currentFlowWrapper, {
+                    style: {
+                        padding: '0',
+                        margin: '0',
+                        pointerEvents: 'none',
+                    },
+                    pixelRatio: 1 / exportZoom,
+                    width: paddedBoundingBox.width * exportZoom,
+                    height: paddedBoundingBox.height * exportZoom,
+                    filter: (node: unknown) => {
+                        if (
+                            node instanceof HTMLElement &&
+                            (node.classList.contains('react-flow__minimap') ||
+                                node.classList.contains('react-flow__controls'))
+                        ) {
+                            return false;
+                        }
+
+                        return true;
+                    },
+                })
+                    .then((dataUrl: string) => {
+                        currentReactFlowInstance.setViewport(oldViewport);
+                        downloadImage(dataUrl, 'chaiNNer-Viewport.png');
```

Comment:
Thanks for this!
What's your opinion on putting the chain name + the current timestamp in the file name?

Reply:
Good idea, will add that